EIPS/eip-5727.md

```diff
@@ -0,0 +1,596 @@
+---
+eip: 5727
+title: Semi-Fungible Soulbound Token
+description: An interface for soulbound tokens, also known as badges or account-bound tokens.
+author: Austin Zhu (@AustinZhu), Terry Chen <terry.chen@phaneroz.io>
+discussions-to: https://ethereum-magicians.org/t/eip-5727-semi-fungible-soulbound-token/11086
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-09-28
+requires: 165
+---
+
+## Abstract
+An interface for soulbound tokens (SBT), which are non-transferable tokens representing a person's identity, credentials, affiliations, reputation, and private assets.
+
+Our interface can handle a combination of fungible and non-fungible tokens in an organized way. It provides a set of core methods that can be used to manage the lifecycle of soulbound tokens, as well as a rich set of extensions that enables DAO governance, privacy protection, token expiration, and account recovery.
+
+This interface aims to provide a flexible and extensible framework for the development of soulbound token systems.
+
+## Motivation
+The Web3 ecosystem nowadays is largely dominated by highly-financialized tokens, which are designed to be freely transferable and interchangeable. However, there are many use cases in our society that requires non-transferablity. For example, a membership card guarantees one's proprietary rights in a community, and such rights should not be transferable to others.
+
+We have already seen many attempts to create such non-transferable tokens in the Ethereum community. However, most of them rely heavily on NFT standards like [EIP-721](./eip-721.md), which are not designed for non-transferability. Others lack the flexibility to support both fungible and non-fungible tokens and do not provide extensible features for critical use cases.
+
+Our interface can be used to represent non-transferable ownerships, and provides features for common use cases including but not limited to:
+
+- granular lifecycle management of SBTs (e.g. minting, revocation, expiration)
+- management of SBTs via community voting and delegation (e.g. DAO governance, operators)
+- recovery of SBTs (e.g. switching to a new wallet)
+- token visibility control (e.g. private SBTs, hiding negative tokens)
+- fungible and non-fungible SBTs (e.g. membership card and loyalty points)
+- the grouping of SBTs using slots (e.g. complex reward schemes with a combination of vouchers, points, and badges)
+
+A common interface for soulbound tokens will not only help enrich the Web3 ecosystem but also facilitates the growth of a decentralized society.
+
+## Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+### Core
+```solidity
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/utils/introspection/IERC165.sol";
```

Comment:
I like to see the interface ids calculated and included in this proposal.

Reply:
Added